x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -315,15 +315,22 @@ class AgentPolicyService {
     soClient: SavedObjectsClientContract,
     options: ListWithKuery & {
       withPackagePolicies?: boolean;
+      fields?: string[];
```

Comment:
added the ability to restrict the agent policy fields returned to reduce payload size.